draft-ietf-dnsop-dns-catalog-zones.md

```diff
@@ -254,12 +280,18 @@ coo.<unique-N>.zones.$OLDCATZ 0 IN PTR $NEWCATZ
 
 The PTR RRset MUST consist of a single PTR record.
 More than one record in the RRset denotes a broken catalog zone which MUST NOT be processed (see (#generalrequirements)).
+A clear error message stating which invalid `coo` property caused the broken
+catalog, with the reason why it was considered invalid, MUST be logged.
 
-When a consumer of catalog zone `$OLDCATZ` receives an update which adds or changes a `coo` property for a member zone in `$OLDCATZ` signalling a new owner `$NEWCATZ`, it does *not* migrate the member zone immediately.
+The `coo` property can be implemented in a stateless fashion when a certain order in the steps is adhered to:
 
-This is because the catalog consumer may not have the `<unique-N>` identifier associated with the member zone in `$NEWCATZ` and because name servers do not index Resource Records by RDATA, it may not know whether or not the member zone is configured in `$NEWCATZ` at all.
-It may have to wait for an update of `$NEWCATZ` adding or changing that member zone.
-When a consumer of catalog zone `$NEWCATZ` receives an update of `$NEWCATZ` which adds or changes a member zone, *and* that consumer had the member zone associated with `$OLDCATZ`, *and* there is a `coo` property of the member zone in `$OLDCATZ` pointing to `$NEWCATZ`, *only then* it will reconfigure the member zone with the for `$NEWCATZ` preconfigured settings.
+   1. First the `coo` property is added to a member zone of `$OLDCATZ`. 
+      Or, instead of adding the `coo` property, an existing property can also be updated with the new value: `$NEWCATZ`.
```

Comment:
It is not defined what value the `coo` property has when it is added (only when it is changed).

```suggestion
   1. First the `coo` property for a member zone of `$OLDCATZ` is set to the value `$NEWCATZ`.
      This is achieved by either creating (or replacing) the corresponding PTR record.
```